istio/datadog_checks/istio/istio.py

```diff
@@ -0,0 +1,42 @@
+# (C) Datadog, Inc. 2010-2018
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+import re
+import traceback
+from contextlib import closing, contextmanager
+from collections import defaultdict
+
+# 3p
+
+# project
+from config import _is_affirmative
+from checks import AgentCheck
+
+from . import mesh, mixer
```

Comment:
Could we keep the check in one file like the others?

Reply:
One of the big rationales behind the pip stuff was so that we didn't have to do this anymore. Also, this is three checks, instead of one (I do not think there's another way to do this).

The agent grabs the first AgentCheck subclass it finds in a check. But I do not really trust the first one that's in the file to be the first one found. However, also, I think this is much better. 

I wanna split up a lot of the other checks, too! The mysql check is 1300 lines, it's horrible to make changes in it (see also, postgres, mongo and many more). It would be much better if we could cut it up, instead.

Reply:
I certainly understand the rationale here, refactoring is good :slightly_smiling_face:

However, I was more concerned with the workflow for modifying checks. One of the side effects of this would be that users/we would have to copy the entire package directory instead of just one file.

This is fine, of course, I just didn't know if this was previously discussed.

Reply:
yea it requires a bit more work to test it than it did before -- I don't love that. But I think the tradeoff is okay